### ISSUE TITLE
Whitelist to disable plugin for certain sites

### ISSUE THREAD

**Issue by eNTi** [dillbyrne/random-agent-spoofer#305]
i stumbled upon a major annoyace the other day. i can't use ip spoofing when trying to connect to my syonology disk station on my lan. it's a security feature and i don't want to disable that. i created a whitelist entry for the ip AND the name (because neither worked) only to realize that i had to disable the plugin completely or uncheck the ip spoofing options. it's not really convenient to disable certain options to use certain sites and then turn them back on again when finished on that site.

please make whitelisted entries either disable ALL options (faster solution) or do a fine grained permission setup to disable certain features of the plugin for certain sites. this could be getting confusing pretty quickly.

i was also a bit lost on the functionality of the settings. so many of the options disable certain features of the browser completely instead of just spoofing them if i understand it correctly? i know. we all hate documentation but some basic info would be good imho. listing the options of the plugin is only partially helpful.

**Comment by flyingzebras** [comment]
I totally agree. I disabled this add-on because I use AMO (addons.mozilla.org) a lot, and the default download buttons becomes disabled.

**Comment by eNTi** [comment]
@flyingzebras 
this can be achieved by putting {"url":"addons.mozilla.org","options":["canvas"]} onto the whitelist and having a farily recent "Whitelist Profile" (above "Whiteliste Rules"). the site will tell you which addons are available by looking at your user agent and disables the buttons accordingly if you don't do that.

**Comment by flyingzebras** [comment]
@eNTi Thank you!

Changing ```[{"url":"play.google.com"},{"url":"s.ytimg.com","options":["canvas"]},{"url":"youtube.com","options":["canvas"]}]``` to ```[{"url":"play.google.com"},{"url":"s.ytimg.com","options":["canvas"]},{"url":"youtube.com","options":["canvas"]}], {"url":"addons.mozilla.org","options":["canvas"]}``` gives a red border around the Whitelist Rules text field box.

Do you know what's wrong in the syntax?

**Comment by eNTi** [comment]
`]}], {"url":"addons.mozilla.org","options":["canvas"]}` to `]}, {"url":"addons.mozilla.org","options":["canvas"]}]`

**Comment by flyingzebras** [comment]
@eNTi Thanks.

This works as you described (green border around the text area): ```[{"url":"addons.mozilla.org","options":["canvas"]},{"url":"play.google.com"},{"url":"s.ytimg.com","options":["canvas"]},{"url":"youtube.com","options":["canvas"]}]```

**Comment by flyingzebras** [comment]
https://github.com/dillbyrne/random-agent-spoofer/issues/317

**Comment by dillbyrne** [comment]
@eNTi I can't have  a turn off completely for site x because some of the preferences are global so will apply to all sites. In the next release the toggle system has been reimplemented so you can quickly enable/disable the addon from the context menu if needed (and the preferences will be saved and reapplied when you reenable the addon ) and also the real profile will work with script injection options too.
